packages/sdk-core/tasks/setupTestEnvironment.sh

```diff
@@ -16,4 +16,6 @@ yarn deploy-contracts-local
 cd ../sdk-core
 yarn generate:graphql-types
 yarn generate:abi-files
-yarn generate:web3-types
\ No newline at end of file
+yarn generate:web3-types
+# compile contracts before starting tests to generate typechain folder
+npx hardhat compile
```

Comment:
Tests slower now?

Reply:
yup, it'll take extra time to compile the contract

Reply:
Approximately how much slower?

Reply:
1 pre-throw routing of the disk

Reply:
~20 seconds

Reply:
@0xdavinchee need harder hat

Reply:
we actually timed it, it takes an extra 5s the first time locally, probably ~20s in build and in subsequent local test runs, almost instant because it won't recompile